### ISSUE TITLE
hex parsing (maybe) not supporting lower case hex

### ISSUE THREAD

**Issue by sublimator** [ripple/ripple-binary-codec#26]
See: https://github.com/ripple/ripple-lib/issues/967

can't recall the reasoning behind not allowing it, or if simply an oversight ... 

this lib started as something custom, intended only for a test suite

**Comment by sublimator** [comment]
https://github.com/ripple/ripple-binary-codec/blob/master/src/utils/bytes-utils.js#L18

To support mixed case encodings, as Buffer does for example:
```javascript 
> Buffer.from('aF', 'hex')
<Buffer af>
```

You'd need 4 string -> number lookups:

```javascript
const byteHexLowerLower = char.toLowerCase() + char2.toLowerCase()
const byteHexUpperUpper = char.toUpperCase() + char2.toUpperCase()
const byteHexLowerUpper = char.toLowerCase() + char2.toUpperCase()
const byteHexUpperLower = char.toUpperCase() + char2.toLowerCase()

res[byteHexLowerLower] = byte;
res[byteHexUpperUpper] = byte;
res[byteHexLowerUpper] = byte;
res[byteHexUpperLower] = byte;
```

**Comment by intelliot** [comment]
Fixed by #97
